x-pack/test/functional_with_es_ssl/apps/triggers_actions_ui/maintenance_windows/maintenance_windows_table.ts

```diff
@@ -209,5 +209,86 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
         expect(upcomingList[0].status).to.equal('Upcoming');
       });
     });
+
+    it('should filter maintenance windows by the archived status', async () => {
+      const finished = await createMaintenanceWindow({
+        name: 'finished-maintenance-window',
+        startDate: new Date('05-01-2023'),
+        notRecurring: true,
+        getService,
+      });
+      objectRemover.add(finished.id, 'rules/maintenance_window', 'alerting', true);
+
+      const date = new Date();
+      date.setDate(date.getDate() + 1);
+      const upcoming = await createMaintenanceWindow({
+        name: 'upcoming-maintenance-window',
+        startDate: date,
+        getService,
+      });
+      objectRemover.add(upcoming.id, 'rules/maintenance_window', 'alerting', true);
+
+      const archived = await createMaintenanceWindow({
+        name: 'archived-maintenance-window',
+        getService,
+      });
+      objectRemover.add(archived.id, 'rules/maintenance_window', 'alerting', true);
+      await browser.refresh();
+
+      await pageObjects.maintenanceWindows.searchMaintenanceWindows('window');
+
+      const list = await pageObjects.maintenanceWindows.getMaintenanceWindowsList();
+      expect(list.length).to.eql(3);
+      expect(list[0].status).to.eql('Running');
+
+      await testSubjects.click('table-actions-popover');
+      await testSubjects.click('table-actions-cancel-and-archive');
+      await testSubjects.click('confirmModalConfirmButton');
+
+      await retry.try(async () => {
+        const toastTitle = await toasts.getTitleAndDismiss();
+        expect(toastTitle).to.eql(
+          `Cancelled and archived running maintenance window 'archived-maintenance-window'`
+        );
+      });
+
+      await pageObjects.maintenanceWindows.searchMaintenanceWindows('archived-mw');
+      await testSubjects.click('status-filter-button');
+      await testSubjects.click('status-filter-archived'); // select Archived status filter
+
+      await retry.try(async () => {
+        const archivedList = await pageObjects.maintenanceWindows.getMaintenanceWindowsList();
+        expect(archivedList.length).to.equal(1);
+        expect(archivedList[0].status).to.equal('Archived');
+      });
+    });
+
+    it('paginates maintenance windows correctly', async () => {
+      new Array(12).fill(null).map(async (_, index) => {
+        const mw = await createMaintenanceWindow({
+          name: index + '-pagination',
+          getService,
+        });
+        objectRemover.add(mw.id, 'rules/maintenance_window', 'alerting', true);
+      });
+      await browser.refresh();
+
+      await pageObjects.maintenanceWindows.searchMaintenanceWindows('pagination');
```

Comment:
Same here.

Reply:
Ok, then I would suggest moving the state for the input in `x-pack/plugins/alerting/public/pages/maintenance_windows/components/maintenance_windows_list.tsx` and keeping only the state for the search in this component. Example

```
// x-pack/plugins/alerting/public/pages/maintenance_windows/index.tsx

const [search, setSearch] = useState<string>('');

<MaintenanceWindowsList
  onSearchChange={setSearch}
/>

// x-pack/plugins/alerting/public/pages/maintenance_windows/components/maintenance_windows_list.tsx

const [search, setSearch] = useState<string>('');

<EuiFieldSearch
    incremental={false}
    placeholder={i18n.SEARCH_PLACEHOLDER}
    value={inputText}
    onSearch={onSearchChange} // from props
    onChange={(e) => setSearch(e.target.value)}
/>
```